streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -81,7 +81,9 @@ public RecordCollector(Producer<byte[], byte[]> producer) {
             if (partitions != null)
                 partition = partitioner.partition(record.key(), record.value(), partitions.size());
         }
-        this.producer.send(new ProducerRecord<>(record.topic(), partition, keyBytes, valBytes), callback);
+        this.producer.send(new ProducerRecord<>(record.topic(), partition, record.timestamp(),
```

Comment:
Hmm, we passed the `record.timestamp()` in `SinkNode`, but it seems not sufficient still. And for changelogger, the timestamp is always set to `null`, which is also not correct?

BTW, we are currently creating two `ProducerRecord` for each send, one in the caller and one inside `RecordCollector` for byte arrays. Could we optimize it away in a separate PR?